docs/apis/plugintypes/qbank/filters.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Question bank filters
+tags:
+  - Plugins
+  - Question
+  - qbank
+description: Question bank plugins allow you to define new filters for the question bank view and random question sets.
+documentationDraft: true
+---
+
+<Since
+version="4.3"
+issueNumber="MDL-72321"
+/>
+
+Question bank plugins allow you define additional filters. These can be used when viewing the question bank, and are included
+in the URL so that a filtered view of the question bank can be shared. They are also used when defining the criteria for adding
+random questions to a quiz.
+
+## Creating a new filter condition
+
+A filter condition consists of two parts - the backend "condition" PHP class, and the frontend "filter" JavaScript class.
+
+The "condition" class defines the general properties of the filter - its name, various options, and how it is applied to the
+question bank query.
+The "filter" class defines how the filter is displayed in the UI, and how values selected in the UI are passed back to the condition.
+
+Each new filter condition must define a new "condition" class in the qbank plugin based on `core_question\local\bank\condition`.
+By default this will use the `core/datafilter/filtertype` "filter" class, although this can be overridden too if required.
+
+### Basic example
+
+This outlines the bare minimum required to implement a new filter condition. This will allow you to filter based on a pre-defined
+list of values, selected from an autocomplete field. This assumes that you already have the basic framework of a qbank
+plugin in place. For real-world examples, look for classes that extend `core_question\local\bank\condition`.
+
+Create a `condition` class within your plugin's namespace. For a plugin called `qbank_myplugin` this would look something like:
+
+```php
```

Comment:
I find it's really handy to specify the location of the class in the title, for example:
```suggestion
```php title="question/bank/myplugin/classes/myfilter_condition.php"
```